packages/fxa-event-broker/lib/version.ts

```diff
@@ -0,0 +1,71 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+/*
+ * Return version info based on package.json, the git sha, and source repo
+ *
+ * Try to statically determine commitHash and sourceRepo at startup.
+ *
+ * If commitHash cannot be found from `version.json` (i.e., this is not
+ * production or stage), then an attempt will be made to determine commitHash
+ * and sourceRepo dynamically from `git`. If it cannot be found with `git`,
+ * just show 'unknown' for commitHash and sourceRepo.
+ *
+ * This module may be called as ./dist/lib/version.js, or when testing,
+ * ./lib/version.ts so we need to look for `package.json` and `version.json`
+ * in two possible locations.
+ */
+
+import * as cp from 'child_process';
+import * as path from 'path';
+
+function readJson(filepath: string) {
+  try {
+    return require(filepath);
+  } catch (e) {
+    /* ignore */
+  }
+
+  return;
+}
+
+function getValue(name: string, command: string): string {
+  const value =
+    readJson(path.resolve(__dirname, '..', 'version.json')) ||
+    readJson(path.resolve(__dirname, '..', '..', 'version.json'));
+
+  if (value && value.version) {
+    return value.version[name];
+  }
+
+  let stdout = 'unknown';
+  try {
+    stdout = cp.execSync(command, { cwd: __dirname }).toString('utf8');
+  } catch (e) {
+    /* ignore */
+  }
+
+  return stdout && stdout.toString().trim();
+}
+
+function getVersionInfo(): object {
+  const commit = getValue('hash', 'git rev-parse HEAD');
+  const source = getValue('source', 'git config --get remote.origin.url');
+
+  const packageInfo =
+    readJson(path.resolve(__dirname, '..', 'package.json')) ||
+    readJson(path.resolve(__dirname, '..', '..', 'package.json'));
+
+  if (packageInfo) {
+    return {
+      commit,
+      source,
+      version: packageInfo.version
+    };
+  }
+
+  return {};
+}
+
+export const version = getVersionInfo();
```

Comment:
Typescript corrections welcome ;-)

Reply:
I think the only thing I'd change is define more concretely what the object is composed of. I don't entirely understand why its not certain where the `package.json` is. Because otherwise this seems like it should always return an object with a `commit` , `source`, and `version` property, no?

Reply:
> I don't entirely understand why its not certain where the package.json is

So the tests use `ts-node` and to  the typescript, so the relative positioning is `../../package.json` (or `../../version.json`).

```
`-- package.json
`-- version.json
`-- lib/
    `-- api/
        `-- version.ts
```

When the compiled javascript is run, the relative positioning is `../../../package.json`.

```
`-- package.json
`-- version.json
`-- dist/
    `-- lib/
        `-- api/
            `-- version.js
```

Having this "maybe it's here, but maybe it's there" check is definitely ugly. Perhaps there is a more idiomatic Typescript way to do this. (I suppose if the typescript code was all under `./src` the relative depths would be the same, but rearranging this entire module for the sake of this change wouldn't be worth it.)

Reply:
Could we create a symlink in `dist/` to make the relative path consistent in both cases?

Reply:
A symlink could be a fix for the `package.json` case, although `version.json` doesn't exist in a local build and spat into the docker image as part of the docker build.

Reply:
> Because otherwise this seems like it should always return an object with a commit , source, and version property, no?

I pushed up a change to simplify and define a return type for `getVersion()`, if that looks right.

Reply:
I think ideally the source should move into a `src/` dir, which will make the relative path locations consistent. We can do that later though.